src/app/tests/integration/chip_im_initiator.cpp

```diff
@@ -303,18 +303,21 @@ int main(int argc, char * argv[])
 
     chip::DeviceLayer::PlatformMgr().StartEventLoopTask();
 
-    err = gTransportManager.Init(chip::Transport::UdpListenParameters(&chip::DeviceLayer::InetLayer)
-                                     .SetAddressType(chip::Inet::kIPAddressType_IPv4)
-                                     .SetListenPort(IM_CLIENT_PORT));
+    gTransportManager = chip::Platform::New<chip::TransportMgr<chip::Transport::UDP>>();
+    gSessionManager   = chip::Platform::New<chip::SecureSessionMgr>();
```

Comment:
Why use heap based gTransportManager, gSessionManager instead of global ones make difference?

It is better to Free those singletons if we allocated from heap.

Reply:
See the comment below, these objects are leaked intentionally.